src/Sulu/Bundle/CustomUrlBundle/EventListener/UpdateLocalizationListener.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\CustomUrlBundle\EventListener;
+
+use Sulu\Component\Webspace\Analyzer\RequestAnalyzerInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+/**
+ * Update localization in case of custom-url route.
+ */
+class UpdateLocalizationListener
```

Comment:
Would call that one `RequestAnalyzerLocalizationListener` (wasn't clear to me which localization is updated here at the beginning).